W-17682660-Update-supported-bindings-CP (#213)
mulesoft/docs-apikit: modules/ROOT/pages/apikit-for-asyncapi-supported-bindings.adoc

include::_attributes.adoc[]
endif::[]

APIkit for AsyncAPI supports various bindings to enable message-driven architectures, and parametrized channels to define dynamic and reusable configurations.

== Supported Bindings

=== Anypoint Connector for MQ (Anypoint MQ Connector) 

[%header%autowidthspread,cols="1a,1a,1a,1a,1a"]
|===
| Binding | Publish | Message Listener | Description | Reference

| `Destination`
| `Destination`
| `Destination`
| `Destination` (queue or exchange) name for this channel. It's optional and defaults to the channel name. Specify it if the channel name differs from the actual `destination` name. For example, if the channel name isn't a valid `destination` name in Anypoint MQ.
.2+| See https://github.com/asyncapi/bindings/blob/b1e7736071ccddc6c3d652bb2071cad4549f8a08/anypointmq/README.md[Anypoint MQ Bindings].

|`DestinationType`
|`DestinationType`
|`DestinationType`

> | `DestinationType` (queue or exchange) name for this channel. It's optional and defaults to `queue`. Specify it if `exchange` is needed instead of a `queue`.

Author: ```suggestion
| Type of destination (queue or exchange, or FIFO queue). It's optional and defaults to `queue`. Specifies the messaging model supported by this channel:

* Publish/subscribe
* Point-to-point
* Strict message ordering
```

Author: @ashutoshsharmasf, can you review and approve this PR if the changes are OK?
Thanks!

| `Destination`
| `Destination`
| `Destination`
| `Destination` (queue or exchange) name for this channel. It's optional and defaults to the channel name. Specify it if the channel name differs from the actual `destination` name. For example, if the channel name isn't a valid `destination` name in Anypoint MQ.
.2+| See https://github.com/asyncapi/bindings/blob/b1e7736071ccddc6c3d652bb2071cad4549f8a08/anypointmq/README.md[Anypoint MQ Bindings].

|`DestinationType`
|`DestinationType`
|`DestinationType`
| Type of destination (queue or exchange, or FIFO queue). It's optional and defaults to `queue`. Specifies the messaging model supported by this channel:

* Publish/subscribe
* Point-to-point
* Strict message ordering
|===

=== Kafka Connector

[%header%autowidthspread,cols="1a,1a,1a,1a,1a"]
|===
| Binding | Publish | Message Listener | Description | Reference

| `Key`
| `Key`
| `Key`
| Key associated with the message in this channel. It's optional and defaults to `no key`. Specifies if the message requires a specific key for partitioning or message grouping. 
| See https://github.com/asyncapi/bindings/blob/master/kafka/README.md[Kafka bindings].
|===

> === Solace

Reviewer: Solace and Solace PubSub+ connector both are same. No new bindings added in Solace PubSub+ connector. Please remove the complete Solace block.
As I mentioned in my previous review comment here https://github.com/mulesoft/docs-apikit/pull/213#issuecomment-2630321600, 
Only two new bindings that need to be added are 

1. key - for kafka protocol, supported in publish and message listener.
The message key. reference - https://github.com/asyncapi/bindings/blob/master/kafka/README.md

2. destinationType - for AnypointMQ supported in Publish and Message listener.

Author: OK! Thanks!

| `priority`
| N/A
| N/A
| Priority level of a message. Priority value ranges from 0 to 255, where 0 is the lowest priority and 255 is the highest. A schema object containing the definition of the priority is also valid.

| `DmqEligible`
| N/A
| N/A
| Eligibility of a message to be moved to a dead-message queue (DMQ). Default value is `false`.
----
operations:
  addPerson:
    action: send
    channel:
      $ref: '#/channels/address'
    messages:
      - $ref: '#/channels/address/messages/personEvent'
    bindings:
      solace:
        bindingVersion: 0.4.0
        destinations:
          - destinationType: queue
            queue:
              name: CreatedHREvents
              topicSubscriptions:
                - person/*/created
          - destinationType: queue
            queue:
              name: UpdatedHREvents
              topicSubscriptions:
                - person/*/updated
        timeToLive: 5000
        priority: 120
        dmqEligible: true
----
|===

=== Solace PubSub+ Connector

[%header%autowidthspread,cols="1a,1a,1a,3a,1a"]
|===
| Binding | Publish | Message Listener | Description | Reference

> | 

Reviewer: ```suggestion
| `deliveryMode`
```

| `deliveryMode` 
| `deliveryMode` 
| Solace PubSub+ Connector operates in `direct` or `persistent` mode. The default mode is `persistent`. The `deliveryMode` parameter determines the quality of service for publishing messages. The `Direct` mode offers lower latency but doesn't guarantee message delivery. The `persistent` mode ensures that messages are reliably delivered, even in the event of a broker failure.
.3+| See https://github.com/asyncapi/bindings/blob/master/solace/README.md[Solace bindings].

> |

Reviewer: ```suggestion
|`destinationType`
```

| `destinationType` 
| `destinationType` 
| Solace PubSub+ Connector supports `queue` in APIkit for AsyncAPI. 

> |

Reviewer: ```suggestion
|`topicSubscriptions`
```

include::_attributes.adoc[]
endif::[]

APIkit for AsyncAPI supports various bindings to enable message-driven architectures, and parametrized channels to define dynamic and reusable configurations.

== Supported Bindings

=== Anypoint Connector for MQ (Anypoint MQ Connector) 

[%header%autowidthspread,cols="1a,1a,1a,1a,1a"]
|===
| Binding | Publish | Message Listener | Description | Reference

|

> `Destination`

Reviewer: Destination and DestinationType are the only binding supported for anypointmq. Please remove others.

Author: OK! Thanks!

| N/A
| Schema object that contains the definitions for Anypoint MQ-specific headers (protocol headers). It's optional. This schema must be of `type` object and have a `properties` key. For example, `messageId` and `messageGroupId` are protocol headers of Anypoint MQ.

----
       payload:
          type: object
          properties:
            #...
        correlationId:
          description: correlation ID is specified as a header and transmitted in the Anypoint MQ message properties section
          location:    $message.header#/correlationId
        bindings:
          anypointmq:
            headers:
              type: object
              properties:
                messageId:
                  type: string
            bindingVersion: '0.1.0'

----

| N/A
| `Destination` 
| `Destination` 
| Channel name Anypoint MQ Connector uses by default. You must specify the `destination` (queue or Exchange) name if the channel name differs from the actual `destination` name, or if the channel name isn't a valid `destination` name.
|===

=== Kafka Connector

[%header%autowidthspread,cols="1a,1a,1a,1a,1a"]
|===
| Binding | Publish | Message Listener | Description | Reference

| `Key`

> | `Key`

Reviewer: Key is the only binding supported for Kafka protocol. Please remove others.

Author: OK! Thanks!
